experimental/examples/token/Token.idr

```diff
@@ -36,30 +43,91 @@ getAccountBalance address accounts = case lookup address accounts of
                       Nothing => 0
                       (Just account) => balance account
 
+total getAccountAllowance : (address : Address) -> SortedMap Address Account -> SortedMap Address Nat
+getAccountAllowance address accounts = case lookup address accounts of
+                      Nothing => empty
+                      (Just account) => allowance account
+
 ||| performTransfer transfers tokens from the from address to the dest address.
 ||| @from the address the tokens to be transferred from
 ||| @dest the address the tokens to be transferred to
 ||| @tokens the amount of tokens to be transferred
+||| @storage the current storage
 total performTransfer : (from : Address) -> (dest : Address) -> (tokens : Nat) -> (storage : Storage) -> Either Error Storage
 performTransfer from dest tokens storage =
   let fromBalance = getAccountBalance from (accounts storage)
       destBalance = getAccountBalance dest (accounts storage) in
         case lte tokens fromBalance of
              False => Left NotEnoughBalance
              True =>
-               let accountsStored = insert from (MkAccount (minus fromBalance tokens) (allowance (getAccount from (accounts storage)))) (accounts storage) in
+               let accountsStored = insert from (MkAccount (minus fromBalance tokens) (getAccountAllowance from (accounts storage))) (accounts storage) in
                  Right
                    (record
                      {accounts =
-                       insert dest (MkAccount (destBalance + tokens) (allowance (getAccount dest (accounts storage)))) accountsStored
+                       insert dest (MkAccount (destBalance + tokens) (getAccountAllowance dest (accounts storage))) accountsStored
                      } storage)
 
+-- Note: Nested record field update syntax(which is safer) doesn't seem to work
+||| updateAllowance updates the allowance map of the allower
+||| @allower the address of the owner of the allowance map that is to be updated
+||| @allowee the address of the allowance from the allower that is to be updated
+||| @tokens the amount of allowance approved by the allower to transfer to the allowee
+||| @storage the current storage
+total updateAllowance : (allower : Address) -> (allowee : Address) -> (tokens : Nat) -> (storage : Storage) -> Storage
+updateAllowance allower allowee tokens storage =
+  case tokens of
+    Z => record
+           {accounts =
+              insert -- update allower's allowance map
```

Comment:
Is there a `modify` function we can use here, that allows us to modify the existing account record?

Reply:
Yup sorry about the repeated code here. I've removed some repetition. Lmk if it's ok.